### DISCUSSION TITLE
Folder structure

### DISCUSSION THREAD

**Discussion by bielu** [KevinJump/uSync#462] [General]
Hi Kevin,
Did you ever consider supporting full folder structure in exports for ContentTypes?
I see few scenarios in which it would helpful :D

**Comment by KevinJump** [comment]
Hi, you can set `UseFlatStructure` to false in the setting then it should export the doctypes (well everything if you do it in default) into the folder structure.

```json
{
    "uSync": {
        "Sets": {
            "Default": {
                "HandlerDefaults": {
                    "UseFlatStructure": false
                },
            }
        }
    }
}
```

It use to be this way by default (back in the Umbraco 4 days) - but it can cause more pain, because every time something moves you end up with orphan files that have to be checked for clashes etc (which uSync does internally) - it just makes it slightly more messy and more likely that you will get conflicts during imports.

but if managed well it does work that way - we just defaulted to the simpler route to reduce potential issues for people.

**Reply by bielu** [reply]
@KevinJump in v9+ if I understand it can be setup per handler, as I would love it to setup it for only for contenType, not for Media etc :)

**Reply by KevinJump** [reply]
Yes it can be done per handler too. 

```json
"uSync" :{
    "Sets": {
        "Default": {
            "Handlers": {
                "ContentTypeHandler" : {
                    "Settings" : {
                          "UseFlatStructure": false
                    }
                }
            }
        }
    }
}
```

**Reply by bielu** [reply]
Thanks that all, what i needed know!
